### ISSUE TITLE
Drop support for ruby 1.9.3

### ISSUE THREAD

**Issue by beauby** [rails-api/active_model_serializers#1302]
Official support for ruby 1.9.3 has been dropped almost a year ago.

The benefits for AMS are:
 + possibility to use lazy enumerators
 + probably many other things

**Comment by NullVoxPopuli** [comment]
Yes :+1:

**Comment by KaraAJC** [comment]
I can work on this too!

**Comment by bf4** [comment]
HI @KaraAJC Thanks for looking here.. this one isn't a good one to work on.  Basically, this one is doesn't require any work except an announcement (with major version change) and a change in the CI matrix.  Have you looked at https://github.com/rails-api/active_model_serializers/blob/master/CONTRIBUTING.md#writing-code-and-comments ? There's lotsa stuff that we've labeled that would be much appreciated, and I can help you with anything, as well.

**Comment by beauby** [comment]
@bf4 I disagree it's not a good one to work on. It has to be done, and it's a pretty straight forward PR (as you mentioned, modify the travis config + make an announcement), so it seems to me like a good candidate for a new contributor.

**Comment by NullVoxPopuli** [comment]
@beauby agreed. Ruby 1.9 is no longer supported officially, so a next release of AMS shouldn't have to support 1.9 either

**Comment by bf4** [comment]
We can't until we decide to break compatibility in 0.10

B mobile phone

**Comment by bf4** [comment]
I'm not opposed, we just have to be responsible ro to end users

B mobile phone

**Comment by beauby** [comment]
Let's look at the facts here:
+ PRs requiring ruby > 1.9.3 have been merged into the 0.8 branch
+ ruby 1.8.7 and 1.9.2 EOLed more than a year ago, ruby 1.9.3 EOLed almost a year ago, so anybody who might be using any of the 0.10-RC would almost certainly not use such old versions

We should definitely make an announcement, but IMO it is a no-brainer to drops support for those.

**Comment by bf4** [comment]
Not disagreeing except with the 'now'  part.  

Fwiw rspec and the mail gem still support 1.8 and heroku had made security releases of Ruby.  EOL is an action item, not a requirement for gem authors.

Again, we can't do it now and we don't need to.  I imagine after we release 0.10 we can then remove < 2.2 and bump to 1.0

Relatedly, that why I added Rails 5 to the build matrix, because that we **do** need to support now.

B mobile phone

**Comment by beauby** [comment]
EOL means the ruby core team won't backport any security fix made to more recent versions. Not sure I understand why we could not do it now.

**Comment by bf4** [comment]
> Not sure I understand why we could not do it now.

I mean, we literally could, but it wouldn't be very nice to our end users.  I admit semver for `0.x` makes no guarantees, and we've made breaking changes from `0.8` to `0.9`, so `0.10` wouldn't be terrible.  

I've just been thinking of 0.10 as the last release before its gets more tightly coupled with Rails, and such should be more flexible.

But, it looks like there may be some feature-pressure to drop 1.9.3, since the json schema gem requires >= 2.0 in  https://github.com/rails-api/active_model_serializers/pull/1270#issuecomment-161810656

So, maybe *can't* is too strong, but I'd like to keep balance breaking changes between their benefits and costs.

If we're dropping stuff, Also Rails 4.0 would be great. It would simplify a number of things. :)

**Comment by bf4** [comment]
oboi, [looks like the decision has already been made](https://github.com/rails-api/active_model_serializers/blob/614e349502e060a57550ad2627e1463d6adb52e3/README.md#release-candidate-please-read).  Full steam ahead.

> Currently this is a release candidate. This is **not** backward
compatible with `0.9.0` or `0.8.0`.

My apologies for the delay.

The .travis.yml can be

```diff
rvm:
-  - 1.9.3
  - 2.0.0
  - 2.1
  - 2.2.2
  - ruby-head
  - rbx-2

install: bundle install --path=vendor/bundle --retry=3 --jobs=3
cache:
  directories:
    - vendor/bundle

script:
  - env CAPTURE_STDERR=${CAPTURE_STDERR:-false} bundle exec rake ci

env:
  - "RAILS_VERSION=4.0"
  - "RAILS_VERSION=4.1"
  - "RAILS_VERSION=4.2"
  - "RAILS_VERSION=master"

matrix:
  exclude:
-  - rvm: 1.9.3
-    env: RAILS_VERSION=master
  - rvm: 2.0.0
    env: RAILS_VERSION=master
```

**Comment by joaomdmoura** [comment]
Totally agree into drop this. :+1:

**Comment by bf4** [comment]
We should consider dropping Rails 4.0, too. It requires minitest 4, which is pretty different from minitest 5. If we drop Rails 4.1, we can also drop support for `_render_option_json`. :)

**Comment by maurogeorge** [comment]
Guys I made a pr to close this https://github.com/rails-api/active_model_serializers/pull/1369 it seens to be a consensus to drop support to Ruby 1.9.3.

About the Rails 4.0, I think we can create another issue and PR.

Thanks guys.

**Comment by maurogeorge** [comment]
@KaraAJC sorry I sent a PR before I saw your comment :disappointed: I will close my PR and wait for yours :heart:

**Comment by KaraAJC** [comment]
thanks ya'll, do you use screenhero? would you help me rebase/squash my
commits from my PR so I can push the updated PR?

[image: photo]
*Kara Carrell*
Dev Bootcamp Grad, Web Developer
m:773-382-8220 | e:KaraAJC@gmail.com
<http://us.linkedin.com/in/KaraAJC>  <http://twitter.com/KaraAJC>
<http://about.me/KaraAJC>  <http://github.com/KaraAJC>

**Comment by beauby** [comment]
Hi @KaraAJC, in order to squash your commits, just do `$ git rebase -i HEAD~4` (here 4 is the number of commits you want to squash, starting from the latest one included). If you're unsure how many commits you want to squash, you can look at the `$ git log`. I cannot pair at the moment, but should you have further questions, feel free to join us on our [Slack](https://amserializers.herokuapp.com/)!

**Comment by KaraAJC** [comment]
I know the commands, just a lil turned around with the order on what's
being squashed and what's being reworded for my commit, ill try again and
if i need another pair of eyes, join the slack in a few...

Awesome thanks!

**Comment by bf4** [comment]
Closed by #1360
